seasmon_xr/ops/whit.py

```diff
@@ -536,6 +536,257 @@ def ws2doptvplc(y, nodata, p, lc, out, lopt):
         lopt[0] = 0.0
 
 
+@guvectorize(
+    [(float64[:], float64, float64[:], boolean, int16[:], float64[:])],
+    "(n),(),(m),() -> (n),()",
+    nopython=True,
+)
+def ws2dwcv(y, nodata, llas, robust, out, lopt):
+    """
+    Whittaker filter GCV optimization of S.
+
+    Args:
+        y (numpy.array): raw data array (1d, expected in float64)
+        nodata (double, int): nodata value
+        llas (numpy.array): 1d array of s values to use for optimization
+        robust (boolean): performs a robust fitting by computing robust weights if True
+    """
+    m = y.shape[0]
+    w = numpy.zeros(y.shape, dtype=float64)
+
+    n = 0
+    for ii in range(m):
+        if (y[ii] == nodata) or numpy.isnan(y[ii]) or numpy.isinf(y[ii]):
+            w[ii] = 0
+        else:
+            n += 1
+            w[ii] = 1
+
+    # Eigenvalues
+    d_eigs = -2 + 2 * numpy.cos(numpy.arange(m) * numpy.pi / m)
+    d_eigs[0] = 1e-15
+
+    if n > 1:
+
+        z = numpy.zeros(m)
+        r_weights = numpy.ones(m)
+
+        # Setting number of robust iterations to perform
+        if not robust:
+            r_its = 1
+        else:
+            r_its = 4
+
+        # Initialising lists for writing to
+        robust_gcv = []
+
+        gcv_temp = [1e15, 0]
+        for it in range(r_its):
+            if it > 1:
+                Sopt_Rog_val = robust_gcv[1][1]
```

Comment:
Please see PEP-8 style guide for [function and variable names](https://peps.python.org/pep-0008/#function-and-variable-names) also for variables further down. I would think we make exceptions to it where it makes sense.

Reply:
Thanks for the link, il will change the variables names. (maybe we could keep 'tr_H' or replace it by 'trace_H' as H is referring to a matrix?)

Reply:
Sure, if there's a good reason to "break" the rule you can